src/analysis/processing/qgsalgorithmrandompointsonlines.cpp

```diff
@@ -0,0 +1,264 @@
+  mNumPoints = parameterAsInt( parameters, QStringLiteral( "POINTS_NUMBER" ), context );
+  mMinDistance = parameterAsDouble( parameters, QStringLiteral( "MIN_DISTANCE" ), context );
+  mMaxAttempts = parameterAsInt( parameters, QStringLiteral( "MAX_TRIES_PER_POINT" ), context );
+  mRandSeed = parameterAsInt( parameters, QStringLiteral( "SEED" ), context );
+  mIncludeLineAttr = parameterAsBoolean( parameters, QStringLiteral( "INCLUDE_LINE_ATTRIBUTES" ), context );
+
+  return true;
+}
+
+QVariantMap QgsRandomPointsOnLinesAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+
+  std::unique_ptr< QgsFeatureSource > lineSource( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !lineSource )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  QgsFields fields = QgsFields();
+  fields.append( QgsField( QStringLiteral( "rand_point_id" ), QVariant::LongLong ) );
+  if ( mIncludeLineAttr )
+    fields.extend(lineSource->fields());
+
+  QString ldest;
+  std::unique_ptr< QgsFeatureSink > sink( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, ldest, fields, QgsWkbTypes::Point, mCrs ) );
+  if ( !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+  //initialize random engine
+  srand(mRandSeed);
+
+  //index for finding nearest neighbours (mMinDistance > 0)
+  QgsSpatialIndex index = QgsSpatialIndex();
+
+  int totNPoints = 0;
+  int missedPoints = 0;
+  int missedLines = 0;
+  int emptyOrNullGeom = 0;
+
+  int featureCount = 0;
+  QgsFeature lFeat;
+  QgsFeatureIterator fitL = lineSource->getFeatures();
+  while ( fitL.nextFeature( lFeat ) )
+  {
+    featureCount++;
+    if ( feedback->isCanceled() )
+    {
+      feedback->pushInfo("cancelled");
+      break;
+    }
+    if ( not lFeat.hasGeometry() )
+    {
+      // Increment invalid features count
+      feedback->pushInfo("No geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    QgsGeometry lGeom( lFeat.geometry() );
+    if (lGeom.isNull())
+    {
+      // Increment invalid features count
+      feedback->pushInfo("Null geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    if (lGeom.isEmpty())
+    {
+      // Increment invalid features count
+      feedback->pushInfo("Empty geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    float lineLength = lGeom.length();
+    int distCheckIterations = 0;
+    int i = 0;
+    while ( i < mNumPoints and distCheckIterations < mMaxAttempts )
+    {
+      if ( feedback->isCanceled() )
+      {
+        break;
+      }
+
+      float randPos = lineLength * (float) rand() / RAND_MAX;
+      QgsGeometry rpGeom = QgsGeometry( lGeom.interpolate( randPos ) );
+      //if ( not rpGeom.isNull() and not rpGeom.isEmpty() )
+      if ( rpGeom.isNull() or rpGeom.isEmpty() )
```

Comment:
```suggestion
      if ( rpGeom.isNull() )
```

(it won't ever be empty)